custom_components/tahoma/tahoma_entity.py

```diff
@@ -3,42 +3,20 @@
 import re
 from typing import Any, Dict, Optional
 
-from homeassistant.const import ATTR_BATTERY_LEVEL
 from homeassistant.helpers.entity import Entity
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 from pyhoma.models import Command, Device
 
 from .const import DOMAIN
 from .coordinator import TahomaDataUpdateCoordinator
 
-ATTR_RSSI_LEVEL = "rssi_level"
-
-CORE_AVAILABILITY_STATE = "core:AvailabilityState"
-CORE_BATTERY_STATE = "core:BatteryState"
 CORE_MANUFACTURER = "core:Manufacturer"
 CORE_MANUFACTURER_NAME_STATE = "core:ManufacturerNameState"
 CORE_MODEL_STATE = "core:ModelState"
 CORE_PRODUCT_MODEL_NAME_STATE = "core:ProductModelNameState"
-CORE_RSSI_LEVEL_STATE = "core:RSSILevelState"
-CORE_SENSOR_DEFECT_STATE = "core:SensorDefectState"
```

Comment:
We need this one. I just checked, and it is used when there is no more battery. For instance:

![image](https://user-images.githubusercontent.com/8216238/128307636-5cb772ec-2915-49b8-ab11-c72e0d2ed462.png)

```JSON
{
	"states": [
		{
			"name": "core:StatusState",
			"type": 3,
			"value": "unavailable"
		},
		{
			"name": "core:SensorDefectState",
			"type": 3,
			"value": "dead"
		},
		{
			"name": "core:DiscreteRSSILevelState",
			"type": 3,
			"value": "verylow"
		},
		{
			"name": "core:RSSILevelState",
			"type": 2,
			"value": 4
		},
		{
			"name": "core:LuminanceState",
			"type": 2,
			"value": 110
		}
	]
}
```

Or perhaps do you plan to implement a battery sensor ?

Reply:
Good to know! I didn't know that this was an important state. The issue is that battery state was coming from multiple properties...

What would you suggest? Add them as multiple sensors or create a separate battery sensor entity? My preference is to have multiple sensors, if the three states are not available at the same time in a device. Than 3 different sensors make sense. However I don't want to add 2 battery sensors accidentally for a device.

Reply:
I think a battery sensor will make more sense, and reduce code complexity. And it will avoid as you said to have multiple sensors linked to the same topic.

Reply:
Agree! However, are we sure that the battery state can exist multiple times?

Reply:
@tetienne should we create an issue for this and do it in a follow up PR? Let's get this huge PR merged asap.

Reply:
Yep, I agree. Just roll back the battery code to be sure we still have the battery logic somewhere.

Reply:
Done.